lib/elixir/lib/map_set.ex

```diff
@@ -18,8 +18,44 @@ defmodule MapSet do
   @opaque t :: %__MODULE__{map: map}
   defstruct map: %{}
 
+  @spec new :: t
   def new(), do: %MapSet{}
 
+  @doc """
+  Creates a `MapSet` from an enumerable.
+
+  Duplicated entries are removed.
+
+  ## Examples
+
+      iex> MapSet.new([:b, :a, 3])
+      #MapSet<[3, :a, :b]>
+
+      iex> MapSet.new([3, 3, 3, 2, 2, 1])
+      #MapSet<[1, 2, 3]>
+
+  """
+  @spec new(Enum.t) :: t
+  def new(enumerable) do
+    Enum.reduce(enumerable, %MapSet{}, &put(&2, &1))
+  end
+
+  @doc """
+  Creates a `MapSet` from an enumerable via the transformation function.
```

Comment:
`tranform` refers to parameter name, while 'the transformation function' is description of the functionality. It follows pattern in `Keyword` description, see https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/keyword.ex#L80 .